scipy/spatial/distance.py

```diff
@@ -1157,7 +1348,7 @@ def sokalsneath(u, v):
 _TEST_METRICS = {'test_' + name: globals()[name] for name in _METRICS.keys()}
 
 
-def pdist(X, metric='euclidean', p=None, w=None, V=None, VI=None):
+def pdist(X, metric='euclidean', *args, **kwargs):
```

Comment:
This is annoying, I thought this PR and gh-7629 were orthogonal. Instead, this is a superset of gh-7629. Let's deal with that one first then.